src/EnergyPlus/EarthTube.cc

```diff
@@ -543,6 +631,188 @@ void CheckEarthTubesInZones(EnergyPlusData &state,
     }
 }
 
+void initEarthTubeVertical(EnergyPlusData &state)
```

Comment:
It would be really great if this function did not loop over all the earth tubes.  If this could just be a one-time init flag on each earth tube, it would be preferable.  I actually don't think it's too much of an effort to change, but I'm open to discussion.

Reply:
That could be done.  The loop over all earth tubes could happen one level up with the continue for a non-vertical model one happening at that point.  It would eliminate the need for the multiple "continues" in every loop and would probably eliminate the need for loops in the initEarthTubeVertical subroutine.  I'll take a look at this when I make other changes requested above.

Reply:
Honestly, if it's working, I would just leave it alone.  We have a new way to call each plant object "one time".  This is nice because then the components can avoid having to constantly check "if one time ... if one time".  As such, plant components definitely shouldn't be looping over each plant instance to do one time inits.  I am a little foggy it seems, and I wanted to apply that same plant logic here to the earth tubes.  But it's not necessary.  Just leave this, thanks.

Reply:
Ok.  It is working so I'm also happy to leave it alone.